aws/src/database/multi_table/update_cautionary_alerts.py

```diff
@@ -0,0 +1,60 @@
+
+from mypy_boto3_dynamodb.service_resource import Table
+
+from aws.src.database.dynamodb.utils.get_dynamodb_table import get_dynamodb_table
+from aws.src.utils.csv_to_dict_list import csv_to_dict_list
+from enums.enums import Stage
+from dateutil import parser
+from datetime import datetime
+
+STAGE = Stage.HOUSING_PRODUCTION
+
+def update_cautionary_alert_in_db_if_tenure_inactive(person_table: Table, alerts_from_csv: list[dict]):
+    """
+    Iterate through the alerts and remove property details if tenure is inactive
+    1. Get person details from dynamo using mmh_id in csv
+    2. check if there is an end date within the tenure object
+    3. if so remove the property details from the csv
+    """
+
+    for alert_item in alerts_from_csv:
+        alert_id = alert_item["id"]
+        mmh_id = alert_item["mmh_id"]
+        if mmh_id is '' or None:
+            alert_item["failed_reason"] = f"mmh_id is null for id {alert_id}"
+            continue
+        person = person_table.get_item(Key={"id": mmh_id}).get("Item")
+
+        person_id = person["id"]
+        if person is None:
+            alert_item["failed_reason"] = f"Could not get person for personId {person_id}"
+            continue
+        tenures = person["tenures"]
+        for tenure_in_person in tenures:
+            end_date = tenure_in_person["endDate"]
+            if end_date is None:
+                continue
```

Comment:
May not apply here, but I figured I'll mention it for the heck of it...
In UH or whatever is replacing it right now, ...
the lack of `end_date` is sometimes represented as `1900-01-01`.
So 1900s dates represent active tenures.

Are we sure dynamo doesn't have the 1900s end of tenure dates? _(from some old times of UH --> Dynamo data migration.. if there was one)_

Reply:
I don't think that is the case on Dynamodb

Reply:
So long you're certain that's okay. We want to mess with Cautionary Alerts on active tenures.